File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}
    np = len(participant)
    nc = len(completion)
    
    for i in range(np):
        if participant[i] not in dic:
            dic[participant[i]] = 1
        else:
            dic[participant[i]] += 1
    
    for j in range(nc):
        dic[completion[j]] -= 1
        
    for k,v in dic.items():
        if v == 1:
            answer = k
            break
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> run = new HashMap<String, Integer>();
        for (int i=0; i<participant.length; i++) {
            run.put(participant[i],run.getOrDefault(participant[i],0)+1);
        }
        for (int j=0; j<completion.length; j++){
            run.put(completion[j],run.get(completion[j])-1);
        }
        for(String key : run.keySet()){
            if(run.get(key) == 1){
                answer += key;
                break;
            }
        }
        return answer;
    }
}